public/app/plugins/panel/table/transformers.ts

```diff
@@ -209,8 +211,11 @@ transformers['json'] = {
       const tableCol: any = { text: column.text };
 
       // if filterable data then set columns to filterable
-      if (data.length > 0 && data[0].filterable) {
-        tableCol.filterable = true;
+      if (data.length > 0) {
+        const dataFrame = toDataFrame(data[0]);
```

Comment:
Don't you convert it to DataFrame already in the toLegacy function (convertJSONDocumentDataToDataFrame)

Reply:
Can you add some more information, do you want me to enforce the type of `data` argument to be an array of data frames?

Reply:
Sorry, confused, just not sure why we convert it back to DataFrame here? Seems like we convert it to toDataFrame (in runRequest) -> legacy (in MetricsPanelCtrl), then here back to DataFrame. 

Do we need this conversion?

Reply:
Hopefully fixed now